src/protocols/secure_channel/CheckinMessage.cpp

```diff
@@ -0,0 +1,110 @@
+/*
+ *    Copyright (c) 2020 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+/**
+ *    @file
+ *      This file implements the Matter Checkin protocol.
+ */
+
+#include "CheckinMessage.h"
+#include <lib/core/CHIPCore.h>
+
+namespace chip {
+namespace Protocols {
+namespace SecureChannel {
+
+static constexpr uint16_t sMinPayloadSize =
+    CHIP_CRYPTO_AEAD_NONCE_LENGTH_BYTES + sizeof(uint32_t) + CHIP_CRYPTO_AEAD_MIC_LENGTH_BYTES;
```

Comment:
It might be clearer if we just did `using CounterType = uint32_t` so it's clear where things are coming from, but either way.